vscode/src/qirGeneration.ts

```diff
@@ -22,44 +22,52 @@ export class QirGenerationError extends Error {
 }
 
 export async function getQirForActiveWindow(
-  supports_adaptive?: boolean, // should be true or false when submitting to Azure, undefined when generating QIR
+  targetSupportsAdaptive?: boolean, // should be true or false when submitting to Azure, undefined when generating QIR
 ): Promise<string> {
   let result = "";
   const program = await getActiveProgram();
   if (!program.success) {
     throw new QirGenerationError(program.errorMsg);
   }
-
+  const isLocalQirGeneration = targetSupportsAdaptive === undefined;
   const config = program.programConfig;
   const targetProfile = config.profile;
-  const is_unrestricted = targetProfile === "unrestricted";
-  const is_base = targetProfile === "base";
+  const isUnrestricted = targetProfile === "unrestricted";
+  const isUnsupportedAdaptiveSubmissionProfile =
+    targetProfile === "adaptive_rif";
+  const targetProfileIsBase = targetProfile === "base";
+  const isSubmittingAdaptiveToBaseAzureTarget =
+    !targetProfileIsBase && targetSupportsAdaptive === false;
 
   // We differentiate between submission to Azure and on-demand QIR codegen by checking
   // whether a boolean value was passed for `supports_adaptive`. On-demand codegen does not
   // have a target, so support for adaptive is unknown.
-  let error_msg =
-    supports_adaptive === undefined
-      ? "Generating QIR "
-      : "Submitting to Azure ";
-  if (is_unrestricted) {
+  let error_msg = isLocalQirGeneration
+    ? "Generating QIR "
+    : "Submitting to Azure ";
+  if (isUnrestricted) {
     error_msg += "is not supported when using the unrestricted profile.";
-  } else if (!is_base && supports_adaptive === false) {
+  } else if (isSubmittingAdaptiveToBaseAzureTarget) {
     error_msg +=
       "using the Adaptive_RI profile is not supported for targets that can only accept Base profile QIR.";
   }
 
   // Check that the current target is base or adaptive_ri profile, and current doc has no errors.
-  if (is_unrestricted || (!is_base && supports_adaptive === false)) {
+  if (
+    isUnrestricted ||
+    isSubmittingAdaptiveToBaseAzureTarget ||
+    isUnsupportedAdaptiveSubmissionProfile
```

Comment:
this should blocks generating QIR for adaptive RIF even outside of submission, which I don't think was the goal. You want something with one more part of the condition:
```suggestion
    (isUnsupportedAdaptiveSubmissionProfile && !isLocalQirGeneration)
```